internal/pkg/utils/testproject/project.go

```diff
@@ -61,14 +75,20 @@ func GetTestProject(t *testing.T, envs *env.Map) *Project {
 	// Load services
 	index, err := storageapi.IndexRequest().Send(p.ctx, p.storageApiClient)
 	if err != nil {
-		assert.FailNow(p.t, "cannot get services: ", err)
+		return nil, func() {
+			cancelFn()
+			unlockFn()
+		}, fmt.Errorf("cannot get services: %w", err)
```

Comment:
Tu `cleanup` funkciu by si mohl ulozit do nejakej premennej a tu a nizsie ju uz iba vratit.
Otazka este je, ci by si ju v pripade chyby nemal rovno zavolat, a vratit prazdnu funkciu.

Lebo, tak ako to pouzivas je to IMHO logicke:
![image](https://user-images.githubusercontent.com/19371734/185605854-4ca44992-0669-47d2-89e1-834c17ccddaf.png)

Ale teda v pripade chyby sa projekt neodomkne (?)

Reply:
Aha. 🙂 No já nad tím přemýšlel tak, že k té chybě mohlo dojít ještě před tím než se vrátila ta `unlockFn`. 🤔 Tak bych ji asi měl tady volat jen pokud není `nil`.

Reply:
Takhle: 
![CleanShot 2022-08-22 at 11 36 35](https://user-images.githubusercontent.com/215660/185890258-edb2166a-da71-4cb4-8192-2c8fab034a55.jpg)

Řeší to všechny tvoje připomínky? 😅 Tomu ukládání `cleanup` funkce do proměnné, resp. jeho motivaci, totiž asi moc nerozumím.

Reply:
> esp. jeho motivaci, 
no motivacia je nemat duplicitny kod ... ak sa bude nieco v tej funkcii menit, tak je tam moznost, ze to niekto zabudne zmenit v ostatnych kopiach

Reply:
Jo počkat, ty myslíš asi aby se tady v `testproject/project` nemuselo všude volat `cancelFn` a `unlockFn` ale zavolala se jen jedna funkce (která je zavolá obě), že? Já myslel že se to nějak týká toho použití zvenku v `operation`. Tak tomu rozumím, jasně.

Reply:
> zavolala se jen jedna funkce

jj, tu funkciu tam vlastne mas, len je nakopirovana viac krat

Reply:
Ok, fixnuto: https://github.com/keboola/keboola-as-code/pull/752/commits/3862d3786deb013fe0a65308b45258d5aca5d01d

Reply:
Super :+1: 

, k tomu `unlockFn` v pripade chyby by sme si asi mohli na chvilku zavolat, nech sa lepsie pochopime (ked budes mat cas)

Reply:
Tak tady: https://github.com/keboola/keboola-as-code/pull/752/commits/3c1fe235d5cfc48100b2aabc65cd0895219b5ee2

Reply:
:+1: